### DISCUSSION TITLE
Filter expressions

### DISCUSSION THREAD

**Discussion by rik1301** [pawelsalawa/sqlitestudio#3830] [Q&A]
Hi, I don't see anything in the wiki but is there a way to use filter expressions in the filter boxes like in DB4S ?  Eg. using % as a wildcard?  At the moment if I search 123 I get all records that contain 123 anywhere in the cell, which is annoying when I only want to see records where those cells end in 123.  In DB4s I can use %123 and it will only show the records ending in 123.

**Comment by pawelsalawa** [comment]
Switch filtering mode to:
![image](https://user-images.githubusercontent.com/481846/89894274-1df26600-dbda-11ea-9fca-4eed6997e3bc.png)
and you can filter as you would write "WHERE" clause, so for example:
```
columnA = 'dsg' and columnB like '%abc%'
```

If you want to just search by mask in any column, then you probably need to use Regular Expression mode and use RegExp syntax, like: `.*123` (the `.*` in RegExp is same as `%` in SQL).

**Comment by rik1301** [comment]
OK.  I'm not sure I understand.  I don't wish to use the main filter box as I have 500,000 records and the '123' is contained in many columns.  I only want to filter in a specific column.  So I have set 'filter by the regular expression' option, and then chosen the 'show filter inputs per column' option to give me the column filters, but when I search .*123 it gives me all cells containing 123, regardless of the location within the cell.  How can I search in the column header boxes for wildcards in a specific format?  Eg. 123% to show all records starting 123, like 123000, 123001, 123999 etc, but not 000123, 999123, 500123 ?  I have tried 123.* and 123*. but neither of these work.  Please help!

I should perhaps clarify that this is for the 'data' tab and 'grid view' tab.

**Comment by pawelsalawa** [comment]
Then you can either go with my first advise (filter by SQL Expression) and type in search box `your_column like '%123'`, or you can enable per-column filter inputs:
![image](https://user-images.githubusercontent.com/481846/89896211-5c3d5480-dbdd-11ea-9603-8739fcefb7f7.png)

**Comment by rik1301** [comment]
I'm not sure you understand the problem.  I *already have* the filter inputs by column activated.  The problem is that the filter wildcards do not work.  Please advise by using the COLUMN FILTER INPUTS how I can use wildcards to only show records where the data is in a specific format.  For example I only want to show records STARTING with 123, NOT containing 123 somewhere else in the field.  I have tried the . and asterisk combinations but it isn't functioning as directed.  I have the 'filter by the Regular Expression' option set in the drop down.  What am I doing wrong?

**Comment by pawelsalawa** [comment]
Use RegExp and type `123.*` (dot before asterisk):
![image](https://user-images.githubusercontent.com/481846/89899040-fa331e00-dbe1-11ea-8fa0-d47348820c45.png)
If you want to do it just for one column, you will need to go with SQL Expression and type exactly which column you want to validate, just like you were wrinting "WHERE" clause, meaning `columnName like '123%'`:
![image](https://user-images.githubusercontent.com/481846/89899199-323a6100-dbe2-11ea-8034-6b26aec923ff.png)

**Comment by rik1301** [comment]
OK.  This is extremely convulted and time-consuming.  It is completely impractical when editing large amounts of data to be typing all that out when 123% in the column top filter should perform that function in a split second.  It should be added as an 'enhancement' as it's a basic requirement for a front-end grid editor.

**Comment by pawelsalawa** [comment]
Please create enhancement proposal ticket then.
